src/main/python/deep_qa/layers/entailment_models/tuple_match.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any, Dict
+
+from keras import backend as K
+from keras.layers import Layer, Dense
+
+from overrides import overrides
+from ...tensors.backend import switch
+
+
+class TupleMatch(Layer):
+    r"""
+        This layer takes as input two tensors cprresponding to two tuples, an answer tuple and a background tuple,
+        and calculates the degree to which the background tuple `entails` the answer tuple.  Entailment is
+        determined by generating a set of entailment features from the tuples (the number of
+        entailment_features = number of tuple slots), and then passing these features into a shallow NN to get an
+        entailment score.
+        Each entailment feature is currently made by comparing the corresponding slots in the two tuples and
+        determining the degree of lexical overlap using the formula:
+            :math:`normalized overlap_s = \dfrac{|A_s \cap B_s|}{|A_s|}`
+        where s is the index of the slot, :math:`A_s` is answer tuple slot s and :math:`B_s` is background
+        tuple slot s.
+
+        Inputs:
+            - tuple_1_input (the answer tuple), shape ``(batch size, num_slots, num_slot_words_t1)``,
+              with an optional mask of shape (batch size, num_slots).  Here num_slot_words_t1 is the maximum
+              number of words in each of the slots in tuple_1.
+            - tuple_2_input (the background_tuple), shape ``(batch size, num_slots, num_slot_words_t2)``,
+              with an optional mask of shape (batch size, num_slots). As above, num_slot_words_t2 is the
+              maximum number of words in each of the slots in tuple_2. This need not match tuple 1.
+
+        Output:
+            - entailment score, shape ``(batch, 1)``
+
+        Parameters
+        ----------
+        params: dict
+            Specifies each of the following (with defaults if unspecified):
+            - num_entailment_features : int, default=4
```

Comment:
Oh, right, but don't you still need a min in there somewhere?  Oh, nevermind, the number of slots is the same in both sets of tuples.  Right.  Sorry... =)